matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -457,63 +389,28 @@ internal class DefaultCryptoService @Inject constructor(
      */
     override fun getDeviceInfo(userId: String, deviceId: String?): CryptoDeviceInfo? {
         return if (userId.isNotEmpty() && !deviceId.isNullOrEmpty()) {
-            cryptoStore.getUserDevice(userId, deviceId)
+            runBlocking {
```

Comment:
we should avoid using runBlocking, exceptions could crash app. We used it mainly in unit test.
Can getDevice throw?

Reply:
How else would you convert a suspendable function into a blocking one? Do note that the actual reads are happening on a rust thread-pool and mostly cached. 

It can throw if there is a error with the store, though that was true for the old cryptostore as well and I'm not sure if that's propagated up the stack.

Reply:
I guess the proper solution is to make `fun getDeviceInfo(...)` suspendable as well. Same for the other methods. I can help to make this change, please let me know @poljar

Reply:
Well sure, but that changes the public API and my intention was to keep changes to the public API minimal for the first pass so we don't introduce bugs because of API changes.